sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -109,7 +109,7 @@ trait CheckAnalysis extends PredicateHelper {
 
         operator transformExpressionsUp {
           case a: Attribute if !a.resolved =>
-            val from = operator.inputSet.map(_.qualifiedName).mkString(", ")
+            val from = operator.inputSet.toSeq.map(_.qualifiedName).mkString(", ")
```

Comment:
Ur, we should not do this in `[TEST]` PR because this is `sql/catalyst/src/main`. Please revert this and make another PR for this if needed.

Reply:
This minor fix is to improve the error messages. It does not worth a separate PR. I removed [TEST]

Reply:
I updated the PR description.

Reply:
Also, this is to fix the non-deterministic test failure in aggregate-part1.sql

Reply:
Got it. Now, this becomes non `[TEST]` and this fix is described in the PR description. Looks good.
> This PR also fixes the error message when the column can't be resolved.